app/models/user.rb

```diff
@@ -228,6 +228,18 @@ def mfa_api_authorized?(otp)
     otp_verified?(otp)
   end
 
+  def mfa_required?
+    return false if mfa_enabled?
+    owner_of_most_downloaded_gem?
+  end
+
+  # NOTE:
+  # Do we want to differentiate the concept of mfa required for an acct where mfa is not enabled vs.
+  # Identifying which accounts require mfa, regardless whether it's enabled or not
+  def owner_of_most_downloaded_gem?
+    (rubygems & Rubygem.by_downloads.limit(Rubygem::MOST_DOWNLOADED)).any?
```

Comment:
I'm having a hard time wrapping my head around this line. It is probably good to add a code comment to explain it a bit.